tests/syslog/test_syslog.py

```diff
@@ -1,62 +1,66 @@
-import time
 import logging
 import pytest
+
 from tests.common.helpers.assertions import pytest_assert
+from tests.common.utilities import wait_until
 
 logger = logging.getLogger(__name__)
 
 pytestmark = [
-    pytest.mark.topology('any')
+    pytest.mark.topology("any")
 ]
 
+
 @pytest.fixture(scope="module")
-def config_syslog_srv(ptfhost):
-    logger.info( "Configuring the syslog server")
+def config_syslog_srv(ptfhost, duthost):
+    logger.info("Configuring the syslog server")
 
-    # add the imudp configuration if not present
-    ptfhost.shell('sed -ni \'/module/!p;$a module(load="imudp")\' /etc/rsyslog.conf')
-    ptfhost.shell('sed -i \'/input(type/!p;$a input(type="imudp" port="514")\' /etc/rsyslog.conf')
+    # Add the imudp configuration if not present
+    ptfhost.shell("sed -ni '/module/!p;$a module(load=\"imudp\")' /etc/rsyslog.conf")
+    ptfhost.shell("sed -i '/input(type/!p;$a input(type=\"imudp\" port=\"514\")' /etc/rsyslog.conf")
 
-    #Remove local /var/log/syslog   
+    # Remove local /var/log/syslog
     ptfhost.shell("rm -rf /var/log/syslog")
 
-    #restart Syslog Daemon 
+    # Restart Syslog Daemon
     ptfhost.shell("service rsyslog restart")
 
-    #Wait a little bit for service to start
-    time.sleep(30)
+    # Wait a little bit for service to start
+    def _is_syslog_running():
+        result = duthost.shell("service rsyslog status | grep \"active (running)\"")["stdout_lines"]
+        return len(result) > 0
+
+    wait_until(30, 3, _is_syslog_running)
 
     yield
```

Comment:
Why do we need yield here if do not perform teardown?

Reply:
We do not, good catch!